react/OrderItems.tsx

```diff
@@ -48,53 +48,70 @@ export const OrderItemsProvider = compose(
     renderComponent(EmptyState)
   )
 )(({ children, ItemListQuery, UpdateItem }: OrderItemsProviderProps) => {
-  const { enqueue } = useOrderManager()
+  const { enqueue, listen } = useOrderManager()
 
   const [itemList, setItemList] = useState(ItemListQuery.cart.items)
 
-  const updateItem = (index: number, quantity: number) => {
-    lastUpdateToken.outdated = true
-    const token = (lastUpdateToken = { outdated: false })
-
-    const updatedList =
-      quantity === 0
-        ? [...itemList.slice(0, index), ...itemList.slice(index + 1)]
-        : adjust(index, item => ({ ...item, quantity }), itemList)
-
-    setItemList(updatedList)
-
-    const task = async () => {
-      const {
-        data: {
-          updateItems: { items },
-        },
-      } = await UpdateItem({
-        variables: {
-          orderItems: [
-            {
-              index,
-              quantity,
-            },
-          ],
-        },
-      })
-
-      if (!token.outdated) {
-        setItemList(items)
+  const isQueueBusy = useRef(false)
+  useEffect(() => {
+    const unlisten = listen('Pending', () => (isQueueBusy.current = true))
+    return unlisten
+  })
+  useEffect(() => {
+    const unlisten = listen('Fulfilled', () => (isQueueBusy.current = false))
+    return unlisten
+  })
+
+  const updateItem = useMemo(
+    () => (index: number, quantity: number) => {
+      const updatedList =
+        quantity === 0
+          ? [...itemList.slice(0, index), ...itemList.slice(index + 1)]
+          : adjust(index, item => ({ ...item, quantity }), itemList)
+
+      setItemList(updatedList)
```

Comment:
That's for optmistic update, right? 

I think we should use the GraphQL API for that instead:
https://www.apollographql.com/docs/react/features/optimistic-ui/

That way, the query will respond immediately

Reply:
I considered this, but doesn't work because the request will be enqueued, so the "optimistic response" will only show up when this task starts running.